docs/specs/restore.yml

```diff
@@ -166,3 +166,94 @@ repos:
         docs/childx.md:
 outputs:
   docs/a.json:
+---
+# restore with dependency lock
+repos:
+  https://docs.com/lock1/a#master:
+    - files:
+        docfx.yml: |
+          dependencies:
+            dep1: https://docs.com/lock1/b#master
+            dep2: https://docs.com/lock1/c
+          dependencyLock: .lock.json
+        docs/a.md: |
+          [!include[](~/dep1/docs/dep1.md)]
+          [!include[](~/dep2/docs/dep2.md)]
+        .lock.json: |
+          {  
+             "git":{  
+                "https://docs.com/lock1/b":{  
+                   "commit":"6a7b7f142f2de52a0b488309f4257d462956f59f",
+                   "git":{  
+                      "https://docs.com/lock1/c#master":{  
+                         "commit":"8720d784fb99abcbdfa922025b8bcc060f0f622a"
+                      }
+                   }
+                }
+             }
+          }
+  https://docs.com/lock1/b:
+    - files:
+        docs/dep1.md: |
+          "abc"
+          [!include[](~/dep2/docs/dep2.md)]
+        docfx.yml: |
+          dependencies:
+            dep2: https://docs.com/lock1/c
+    - files:
+        docs/dep1.md: |
+          "def"
+          [!include[](~/dep2/docs/dep2.md)]
+        docfx.yml: |
+          dependencies:
+            dep2: https://docs.com/lock1/c
+  https://docs.com/lock1/c:
+    - files:
+        docs/dep2.md: "2333"
+    - files:
+        docs/dep2.md: "233333"
+outputs:
+  docs/a.json: |
+    {"content":"<p>&quot;def&quot;</p>\n<p>233333</p>\n<p>2333</p>\n"}
+---
+# restore with sub dependency lock only
```

Comment:
add a test case which contains  both root lock and sub-lock, root lock should dominate

Reply:
done